x-pack/plugins/cases/server/routes/api/cases/get_case.ts

```diff
@@ -30,6 +33,9 @@ export function initGetCaseApi({ router, logger }: RouteDeps) {
         const id = request.params.case_id;
 
         return response.ok({
+          headers: {
```

Comment:
It looks like the warning header will be returned on every call to this endpoint - but don't you just want the header if they use the `query.includeComments` property?

Reply:
Oh, good point! I will update accordingly.

Reply:
@pmuellr I just noticed that we default it to `true` if the user omits it so I don't think we can distinguish if the user provided or not.

Reply:
Ah, bummer.

You could change the type to `boolean | undefined`, and then apply the default outside of the validation, but that's clumsy, and changes your typing which could require more undefined checks.  Perhaps instead, since `false` is the "deprecated" version, you can just check after the validation if it's `false` and then add the header?  So it wouldn't be marked deprecated if they used the param and it was true.

Another possibility would be to see if there's a way to get the parameters in more of a "raw" state, after the validation, from the request.  And then put the check there.  

I really don't think we want this header used **every time** the API is used.

Reply:
@pmuellr Amazing idea about the "raw" state. I pushed the changes you suggested.